### ISSUE TITLE
vertex color incorrect

### ISSUE THREAD

**Issue by ghost** [Befzz/blender3d_import_psk_psa#82]
Blender: 3.0.1

Hello,

I created a .psk and .gltf from the game 'GranBlue' using umodel(fmodel). While .gltf imports clean vertex color for me (right), it is washed out and buggy for .psk (left). Here is a [mega.nz](https://mega.nz/folder/1SZGVTra#tc-r8U8sHbxjaritCKqhhg) link with the .psk and .gltf model.

![vertexColor](https://user-images.githubusercontent.com/44917874/162186482-dfd602e5-257e-4fbe-a3b2-c0491511d30d.png)

Thanks for the addon and your help.

**Comment by yretenai** [comment]
This is because the mesh does not have smooth shading and is instead relying on normals. Go into edit mode and press F and select "Shade Smooth". However, verify that this is how the game behaves.

**Comment by cyso1988** [comment]
(In all images .psk is on the left, .gltf on the right - the quality of my .gif is a bit bad, sorry).

Thank you for your answer, I was afraid that I would not be taken seriously. ^^
Since I think I'm being misunderstood, here's my view:

I create a .psk with any Granblue model with the following versions:
- umodel
- umodel custom version with normals
- fmodel

All three versions, on all models bring incorrect vertex color in Blender. If I manually enable 'Smooth Shading' as suggested, or remove 'Custom Split Data' - it always remains incorrect (also believe that vertex normals are rather unrelated to vertex color). If I use a 'Data Transfer' modifier I can transfer the vertex color from a .glTF mesh to the .psk mesh without changing anything in the shading or the normals - reason more to exclude it (pay attention to the settings).

![switch](https://user-images.githubusercontent.com/44917874/163715048-b440c7fc-b6e8-4fbb-bac1-bf75fe31d739.gif)

Whereas .glTF imports correct vertex color for all models.

I use the ArcSystem shader from Aerthas for the models in Blender, where Vertex Color is used as a mask for permanent shadows.

You can see it quite well in this example. The permanent shadow with .glTF vertex color is correct, .psk vertex color is broken.

![switch2](https://user-images.githubusercontent.com/44917874/163715067-5272796e-6197-4dbc-8c16-9f62fa02ef33.gif)

To me it looks like this:
The vertex color information that the .psk importer writes to the mesh is somehow incorrect.
If it is really only the shading - please make a short .gif and show it to me. I can't reproduce the solution.

I am happy to supply .blends etc if it helps solve the problem.

**Comment by cyso1988** [comment]
What is the reason for this? I don't understand enough about Blender's API.

The console shows me a perpetual list of vertices triggered by the piece of code between 1087 - 1093 when importing.

![console](https://user-images.githubusercontent.com/44917874/163719121-9a4d4a47-adc8-4d5d-b8af-ee36f26df94d.png)

**Comment by yretenai** [comment]
Yeah I see now, it seems FModel (CUE4Parse) and UModel both narrow the number of vertices down by deduplicating according to position and normal.

The GLTF exporter just exports the entire vertex stream.

**Comment by cyso1988** [comment]
Thank you for your time.
If you are the one person who will fix it that would be great of course. :P : )
Stay with glTF for now.

**Comment by floxay** [comment]
Could you please share the raw data? (save packages option in UEViewer)
